### PR TITLE
API to show pkg version based on git tag

### PR DESCRIPTION
**Pull request by rumatoest**

### CODE REVIEW

File: src/factory/build.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }
+        Err(e) => {
+            println!("cargo:warning=Can not extract git tag: {}", e);
+            return;
+        }
+    };
+
+    println!("cargo:warning=GIT tag extracted: {}", tag);
+
+    println!("cargo:rustc-env=CARGO_PKG_VERSION={}", tag);
```
Comment (hagsteel) [inline src/factory/build.rs:27]:
```suggestion
    let git_out = Command::new("git").args(["describe", "--tag"]).output();

    match git_out {
        Ok(o) if o.status.success() => {
            let tag = String::from_utf8_lossy(&o.stdout).trim().to_string();
            println!("cargo:warning=GIT tag extracted: {}", tag);
            println!("cargo:rustc-env=CARGO_PKG_VERSION={}", tag);
        }
        Ok(o) => println!(
            "cargo:warning=Git command exited with error: {}",
            String::from_utf8_lossy(&o.stderr).to_string()
        ),
        Err(e) => println!("cargo:warning=Can not extract git tag: {}", e),
    }```

**Review by hagsteel — CHANGES_REQUESTED** [review]
Clever to use the build.rs here!

File: src/factory/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 [package]
 edition = "2021"
 name = "token-factory"
-version = "0.1.6"
+# Just a fallback value. Valid version will be extracted during build.
+version = "0.0.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (rumatoest) [reply src/factory/Cargo.toml]: Good point

File: src/factory/Cargo.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [package]
 edition = "2021"
 name = "token-factory"
-version = "0.1.6"
+version = "0.24.0" #Will be replaced with the latest tag while building from the git
```
Comment (hagsteel) [inline src/factory/Cargo.toml:4]: This comment makes it sound like the build tool will modify the toml file.
Continuing this conversation in chat

Comment (hagsteel) [inline src/factory/Cargo.toml:4]:
```suggestion
version = "0.24.0"
```

**Review by hagsteel — CHANGES_REQUESTED** [review]

File: src/factory/Cargo.toml (hunk 3 above)
Comment (rumatoest) [reply src/factory/Cargo.toml]: Done

**Review by hagsteel — APPROVED** [review]
good work!

**Review by AlessandroRiet — APPROVED** [review]

**Pull request merged**